website/integrations: grafana: add Helm and Terraform config examples

senare opened the pull request:
<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://goauthentik.io/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->

Ill try and add some example for doing this using terraform, will that be oki ?

Also added helm section

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

Commit:
Added helm ref and Terraform Provider config

   Added disclaimer about secret's in Helm or i.e values.yaml

Co-authored-by: risson <18313093+rissson@users.noreply.github.com>
Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: senare <senare@gmail.com>

rissson commented on website/integrations/services/grafana/index.mdx:
```diff
@@ -26,6 +26,72 @@ Create an application in authentik. Create an OAuth2/OpenID provider with the fo
+}
+
+resource "authentik_scope_mapping" "grafana-login" {
+  name       = "grafana-login"
+  scope_name = "login"
+  expression = "return request.user.username"
+}
+
+resource "authentik_scope_mapping" "grafana-name" {
+  name       = "grafana-name"
+  scope_name = "name"
+  expression = "return request.user.name"
+}
+
+resource "authentik_provider_oauth2" "provider_grafana" {
+  name          = "Grafana"
+  client_id     = "my_client_id"
+  client_secret = "my_client_secret"
+
+  authentication_flow = data.authentik_flow.grafana-authentication-flow.id
+  authorization_flow  = data.authentik_flow.grafana-authorization-flow.id
+
+  redirect_uris = ["https://grafana.company/login/generic_oauth"]
+
+  property_mappings = [
+    authentik_scope_mapping.grafana-login.id,
+    authentik_scope_mapping.grafana-name.id,
+    data.authentik_scope_mapping.oauth-gitea-email.id,
+    data.authentik_scope_mapping.oauth-gitea-profile.id,
+    data.authentik_scope_mapping.oauth-gitea-openid.id,
+  ]
+}
+
+resource "authentik_application" "application_grafana" {
+  name              = "Grafana"
+  slug              = "grafana-slug"
+  protocol_provider = authentik_provider_oauth2.provider_grafana.id
+
+  meta_launch_url = "https://grafana.company/"
+}
```
```suggestion
data "authentik_certificate_key_pair" "default" {
  name = "authentik Self-signed Certificate"
}

data "authentik_flow" "default-provider-authorization-implicit-consent" {
  slug = "default-provider-authorization-implicit-consent"
}

data "authentik_scope_mapping" "scope-email" {
  name = "authentik default OAuth Mapping: OpenID 'email'"
}

data "authentik_scope_mapping" "scope-profile" {
  name = "authentik default OAuth Mapping: OpenID 'profile'"
}

data "authentik_scope_mapping" "scope-openid" {
  name = "authentik default OAuth Mapping: OpenID 'openid'"
}

resource "authentik_provider_oauth2" "grafana" {
  name          = "Grafana"
  # Optional: will be generated if not provided
  # client_id     = "my_client_id"
  # client_secret = "my_client_secret"

  authorization_flow  = data.authentik_flow.default-provider-authorization-implicit-consent.id

  redirect_uris = ["https://grafana.company/login/generic_oauth"]

  property_mappings = [
    data.authentik_scope_mapping.scope-email.id,
    data.authentik_scope_mapping.scope-profile.id,
    data.authentik_scope_mapping.scope-openid.id,
  ]
  
    signing_key = data.authentik_certificate_key_pair.default.id
}

resource "authentik_application" "grafana" {
  name              = "Grafana"
  slug              = "grafana"
  protocol_provider = authentik_provider_oauth2.grafana.id
}
```

Updated the terraform to only have what's needed and to match with the other instructions.

rissson commented on website/integrations/services/grafana/index.mdx:
```diff
@@ -78,6 +145,29 @@ token_url = https://authentik.company/application/o/token/
+    name: authentik
+    enabled: true
+    client_id: "<Client ID from above>"
+    client_secret: "<Client Secret from above>"
```
For Helm values, I don't like telling users to put their client_secret in the values straight away. We should at least add a comment that says that this can also be passed as an environment variable with:
```yaml
envValueFrom:
  GF_AUTH_GENERIC_OAUTH_CLIENT_ID:
    secretKeyRef:
      name: grafana-oauth-credentials
      key: client_id
  GF_AUTH_GENERIC_OAUTH_CLIENT_SECRET:
    secretKeyRef:
      name: grafana-oauth-credentials
      key: client_secret
```

senare replied:
Yes, I agree and the reasoning is sound !

however the reason I put it in there in the first place, and hardcoded values to match (instead of generated) was to make it easy to read and follow. I however have no strong feelings about it either way.

Let me see if I can make the changes, I am afraid I am not accustomed to the 'github part' of 'git' yet

senare replied:
This gets a bit hairy ... I think there are 3 ways of doing this ... and if we pass it as env to Grafana it might break 
i will link the issue with discussion if I can find it again !

here it is [17363](https://github.com/helm/charts/issues/17363) could be old and outdated ...

So i am guessing thats the reason for the Grafana Helm docs to spec that you should add as a file ?

[Grafana Helm docs ](https://github.com/grafana/helm-charts/blob/main/charts/grafana/README.md#how-to-securely-reference-secrets-in-grafanaini)

I will look into it and try it before we doc it ... in my case I dodge the issue using 

#   $ openssl rand -hex 32

which would mean there are spec chars to break the env import, do we have any such guarantees if go with generated values ? Would that be a better option ?

rissson replied:
We're using that way for our grafana instance so I'm pretty sure it works. We can also not specify a specific way of doing things, but just have a comment along the lines of "The client secret should not be passed directly via Helm values, but via a secret. Check Grafana Helm chart documentation for more information."

senare commented:
I don't think we need to spec 'authentik Self-signed Certificate' it would be default as in the other cases ...

I will update and retest my configuration and update after !

rissson commented:
> I don't think we need to spec 'authentik Self-signed Certificate' it would be default as in the other cases ...

You may be right. However, it might be the default on the UI but not in the terraform provider.

senare commented:
Sorry, i don't want to leave this hanging! 

- I updated with a disclaimer would that be good enough for now ?
- Also added groups (Grafana Admins, Grafana Editors, Grafana Editors) as assumed. 
- Removed the cert seams to be fine with default (when I test anyways).
- However i added client_id back in again as terraform provider has it as required.
- Rebase & Squash

Goode enough ?

rissson commented on website/integrations/services/grafana/index.mdx:
```diff
@@ -26,6 +26,67 @@ Create an application in authentik. Create an OAuth2/OpenID provider with the fo
+
+resource "authentik_provider_oauth2" "grafana" {
+  name          = "Grafana"
+  #  Required by should be fine to use
```
```suggestion
  #  Required. You can use the output of:
```

rissson commented on website/integrations/services/grafana/index.mdx:
````diff
@@ -80,6 +142,33 @@ api_url = https://authentik.company/application/o/userinfo/
+
+```
+
+Note: for security reasons you shouldn't inline the client_secret instead you should adopt solution to your security requirements. Maybe you are using Sealed Secrets, SOPS or External Secrets ?
+for more information reference 'https://github.com/grafana/helm-charts/blob/main/charts/grafana/README.md#how-to-securely-reference-secrets-in-grafanaini'
````
```suggestion
:::note
For security reasons you shouldn't inline the client_secret in the values, but use a secret instead. For more information, see https://github.com/grafana/helm-charts/blob/main/charts/grafana/README.md#how-to-securely-reference-secrets-in-grafanaini
:::
```

rissson reviewed (changes requested):
Some rewording but otherwise LGTM

tanberry commented on website/integrations/services/grafana/index.mdx:
```diff
@@ -80,6 +142,34 @@ api_url = https://authentik.company/application/o/userinfo/
 
+  </TabItem>
+  <TabItem value="helm">
+If you are using helm values-file instead, you have to set these options:
```
```suggestion
If you are using a Helm `values.yaml` file instead, you have to set these options:
```

tanberry reviewed (approved):
Thanks for the contribution, @senare !

senare commented:
I did run

$ make website-lint-fix

but it wont say what's not linted nor did it correct it, any advice ?

BeryJu committed:
fix formatting

Signed-off-by: Jens Langhammer <jens@goauthentik.io>
